src/dfcx_scrapi/tools_ml/utterance_generator_utils.py

```diff
@@ -195,46 +238,59 @@ def _generate_phrases(self, training_phrases: pd.DataFrame, set_size: int):
-        """
-        Create a test set for a given list of intents. The phrases in this set will not
-        be exact string match phrases which exist in the training phrases but will be close
-        semantically.This set is automatically labeled by the intent whose training was
-        used to generate the new phrase. This can be used to run through the core.conversations
-        run_intent_detection function. You may need to specify a flow_display_name and
-        page_display_name in the dataframe to run the set at the correct location.
-
+    def create_test_set(
+        self, agent_id: str, intent_subset: list, set_size: int = 100
+    ):
+        """Creates a test set for a given list of intents. 
+        
+        The phrases in this set will not be exact string match phrases which
+        exist in the training phrases but will be close semantically. This set
+        is automatically labeled by the intent whose training was used to
+        generate the new phrase. This can be used to run through the
+        core.conversations run_intent_detection function. You may need to
+        specify a flow_display_name and page_display_name in the dataframe to
+        run the set at the correct location.
 
         Args:
-            agent_id: (string) name parameter of the agent to pull intents from - full path to
-                agent
-            intent_subset: (list) display names of the intents to create a test for, base
-                phrases come from the training in the
+            agent_id: (string) name parameter of the agent to pull intents from
+                full path to agent
+            intent_subset: (list) display names of the intents to create a test
+                for, base phrases come from the training in the intent.
+            set_size: (int) overall target size of the test set to create, may
+                be less depending if new independent phrases can be generated
+                from the data. The function tries to get even entries per
                 intent.
-            set_size: (int) overall target size of the test set to create, may be less
-                depending if new independent phrases can be generated from the data.
-                The function tries to get even entries per intent.
-
-
-
 
         Returns:
-          test_set: (DataFrame) a pandas dataframe consisting of rows of the utterance and
-          the intent which it was generated from. This generated from intent can be used as
-          the true label.
+          test_set: (pd.DataFrame) a pandas dataframe consisting of rows of the
+          utterance and the intent which it was generated from. This generated
+          from intent can be used as the true label.
         """
-        synthetic_set = self.create_synthetic_set(agent_id, intent_subset, set_size)
+        synthetic_set = self.create_synthetic_set(agent_id, intent_subset, 
```

Comment:
@Hgithubacct these internal variables should also be renamed for future debugging.

If `synthetic_set` isn't actually of type `set` let's name it `synthetic_dataset` just to avoid confusion.